### ISSUE TITLE
missing "Null pointer exception may occur here. " with duplicate variable names 

### ISSUE THREAD

**Issue by voku** [kalessil/phpinspectionsea#1552]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Ultimate), 3.0.9  |
| Language level | PHP 7.2                |

Current behaviour
--
```php
<?php

class Lall {
    public function echo() { echo "hello"; }
}
class Foo {
    /** @var Lall|null */
    public $lall;

    public function __construct() {
        $this->lall = mt_rand(0,100) > 50 ? new Lall() : null;
    }

    public function getLall(): ?Lall
    {
        return $this->lall;
    }
}

class FooBar {
    /** @var Lall|null */
    public $lall;

    public function __construct() {
        $this->lall = mt_rand(0,100) > 50 ? new Lall() : null;
    }

    public function getLall(): ?Lall
    {
        return $this->lall;
    }
}

$foo = (new Foo())->getLall();
if ($foo) {
    $foo->echo();
}

// ...

$foo = (new FooBar())->getLall();
$foo->echo();
```

... here I expected a warning that the second "$foo" is still maybe NULL.

![Bildschirmfoto vom 2020-04-29 14-02-21](https://user-images.githubusercontent.com/264695/80593986-76826680-8a22-11ea-8680-058d78f850fe.png)
![Bildschirmfoto vom 2020-04-29 14-02-04](https://user-images.githubusercontent.com/264695/80593991-771afd00-8a22-11ea-9822-3614962f11fa.png)
![Bildschirmfoto vom 2020-04-29 14-01-53](https://user-images.githubusercontent.com/264695/80593993-77b39380-8a22-11ea-9d64-0c3787a9e07a.png)

Expected behaviour
--
show the second "echo()" call as error because it produces a fatal error in php 

Environment details
---
PhpStorm 2020.1
Build #PS-201.6668.153, built on April 13, 2020
Licensed to Portable UTF-8 / Lars Moelleken
Subscription is active until February 27, 2021
For non-commercial open source development only.
Runtime version: 11.0.6+8-b765.25 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Linux 4.4.0-177-generic
GC: G1 Young Generation, G1 Old Generation
Memory: 5120M
Cores: 4
Registry: run.processes.with.pty=TRUE, ide.tooltip.initialDelay=885, ide.balloon.shadow.size=0
Non-Bundled Plugins: com.dubreuia, uk.co.ben-gibson.remote.repository.mapper, Key Promoter X, String Manipulation, com.alayouni.ansiHighlight, com.andrey4623.rainbowcsv, com.intellij.ideolog, com.upsolver.PreCommitHook, dev.arunvelsriram.desccron, mobi.hsz.idea.gitignore, name.kropp.intellij.makefile, net.seesharpsoft.intellij.plugins.csv, net.vektah.codeglance, tanvd.grazi, ru.adelf.idea.dotenv, NEON support, de.espend.idea.php.toolbox, lv.midiana.misc.phpstorm-plugins.deep-keys, com.kalessil.phpStorm.phpInspectionsUltimate, me.artspb.idea.eval.plugin, net.king2500.plugins.PhpAdvancedAutoComplete, izhangzhihao.rainbow.brackets
Current Desktop: Unity
